baseapp/baseapp.go

```diff
@@ -56,23 +59,40 @@ func (b *BaseApp) Logger() log.Logger {
 }
 
 // Start starts the baseapp.
-func (b *BaseApp) Start() {
+func (b *BaseApp) Start(ctx context.Context) error {
 	b.Logger().Info("starting app")
 
-	// TODO: create a new context for every job request / creation.
-	ctx := sdk.NewContext(
-		context.Background(),
+	// Wrap the context in sdk.Context in order to attach our clients, logger and db.
+	// TODO: is this bad practice we are just stealing from the cosmos sdk?
+	ctx = sdk.NewContext(
```

Comment:
we need to really make sure that every call to a job uses a diff underlying context for all .Setup(), .Subscribe()/.EthSubscribe(), and .NewPayload()

Reply:
agree, just haven't gotten there yet.